### ISSUE TITLE
Tube filter misses some points

### ISSUE THREAD

**Issue by GastonPerret** [Kitware/vtk-js#1676]
Hello. I am using the vtkTubeFilter to draw sets of points stored in VTP files. It works fine for most of the points, but at some extremities the tube misses a couple of points.
Im using the Tube Viewer example to test it
[https://kitware.github.io/vtk-js/examples/TubesViewer.html](url)

![image](https://user-images.githubusercontent.com/62260505/103295433-b4c98e80-49d2-11eb-9ca7-e9ca0084b1df.png)
![image](https://user-images.githubusercontent.com/62260505/103295532-f0645880-49d2-11eb-92fc-dcd8cfe445e7.png)

Any idea of what could cause the problem? Thank you!

**Comment by jourdain** [comment]
Could you provide some test data?

**Comment by eltopo22** [comment]
> Could you provide some test data?

Yes, thanks for answering.This is one of the VTP files that shows the problem
[tube error.vtp.zip](https://github.com/Kitware/vtk-js/files/5755089/tube.error.vtp.zip)

**Comment by MHadavand** [comment]
Hi,

I encountered similar issue with the tube filter. For most variables, after wiring the tube filter the data values get changed to a single values and then the data ranged with be zero that causes this error "attempt to set zero width color range".

I tried the online example tube viewer form vtk-js website and got the same error. I attached a vtp file that contains one tube. Only the point data Z gets rendered correctly and for the rest of the variables, the data array gets changed to a single value.


[IntervalData_Trim.zip](https://github.com/Kitware/vtk-js/files/5784479/IntervalData_Trim.zip)

Your help is much appreciated.
Thanks
